### PR TITLE
STITCH-3297: add admin sdk endpoint for rerunning unconfirmed user confirmation flow

### PR DESCRIPTION
**Pull request by kevintian**
JIRA: [STITCH-3297](https://jira.mongodb.org/browse/STITCH-3297)

Driveby: Fixed a few values inside of `userRegistrations.test.js` so that all the tests passed in the file.

### CODE REVIEW

**Comment by mburdette** [comment]
This PR title seems incorrect since you are actually changing the sdk not api

File: test/userRegistrations.test.js [hunk 1]
```diff
@@ -29,20 +29,65 @@ describe('User Registrations', ()=>{
...
+  });
+  it('rerunUserConfirmation should confirm existing pending user', async() => {
+    await client.register(testEmail, password);
+    expect(await findPendingUsersByEmail(testEmail)).toHaveLength(1);
+    let { token_id: tokenId, token } = await th.app().userRegistrations().rerunUserConfirmation(testEmail);
+    await client.auth.provider('userpass').emailConfirm(tokenId, token);
+
+    // We must login first so that a user id is attached to the password record
+    await client.login(testEmail, password);
+    expect(client.auth.loggedInProviderType).toEqual(PROVIDER_TYPE_USERPASS);
```
Comment (christineschen) [inline test/userRegistrations.test.js]: Would this only be true if the email confirmation was successful?

File: src/admin.js [hunk 2]
```diff
@@ -294,6 +294,7 @@ export class StitchAdminClient extends StitchClient {
 
           userRegistrations: () => ({
             sendConfirmationEmail: (email) => api._post(`${appUrl}/user_registrations/by_email/${email}/send_confirm`),
+            rerunUserConfirmation: (email) => api._post(`${appUrl}/user_registrations/by_email/${email}/rerun_confirmation`),
```
Comment (kevintian) [inline src/admin.js]: note: change this to match the `rerun_confirm` change inside of the admin api

File: test/userRegistrations.test.js [hunk 3]
```diff
@@ -29,20 +29,65 @@ describe('User Registrations', ()=>{
...
     expect(await findPendingUsersByEmail(testEmail)).toHaveLength(0);
-    expect(await getPendingUsers()).toHaveLength(1);
+    expect(await getPendingUsers()).toHaveLength(0);
   });
   it('removePendingUserByID should remove existing pending user', async() => {
     await client.register(testEmail, password);
     const userID = (await getPendingUsers())[0]._id;
     expect(await findPendingUsersByID(userID)).toHaveLength(1);
     await th.app().userRegistrations().removePendingUserByID(userID);
     expect(await findPendingUsersByID(userID)).toHaveLength(0);
```
Comment (mburdette) [inline test/userRegistrations.test.js]: can some of these be done inside a `beforeEach`? seems like a bunch of tests do this stuff

File: test/userRegistrations.test.js [hunk 4]
```diff
@@ -76,13 +121,13 @@ describe('User Registrations', ()=>{
     await client.linkWithProvider('userpass', { username: linkEmail, password });
 
     pendingUsers = await th.app().userRegistrations().listPending({limit: '3'});
-    expect(pendingUsers).toHaveLength(2);
+    expect(pendingUsers).toHaveLength(1);
```
Comment (mburdette) [inline test/userRegistrations.test.js]: still not understanding why all these numbers are changing...lol

File: test/userRegistrations.test.js (hunk 1 above)
Comment (kevintian) [reply test/userRegistrations.test.js]: We wouldn't be able to login successfully if the email confirmation was not successful

**Pull request merged**